app/javascript/lib/scp-api.js

```diff
@@ -120,6 +120,67 @@ export function setMockOrigin(origin) {
   mockOrigin = origin
 }
 
+/** Constructs and encodes a URL parameter, if it has a value */
+function encodeParam(param, value, isFirst=false) {
+  const delimiter = isFirst ? '?' : '&'
+  if (!value || value === '') {
+    // Omit empty parameters in API request
+    return ''
+  } else {
+    return `${delimiter}${param}=${encodeURIComponent(value)}`
+  }
+}
+
+/**
+* Returns initial content for the "Explore" tab in Study Overview
+* @param {String} studyAccession Study accession
+*/
+export async function fetchExploreInitialization(studyAccession, mock=false) {
+  const apiUrl = `/studies/${studyAccession}/explore`
+  const [exploreInit, perfTime] =
+    await scpApi(apiUrl, defaultInit(), mock, false)
+
+  return exploreInit
+}
+
+/**
+ * Returns an object with scatter plot data for a cluster in a study
+ *
+ * This endpoint is volatile, so intentionally not documented in Swagger.
+ *
+ * see definition at: app/controllers/api/v1/visualization/clusters_controller.rb
+ *
+ * @param {String} studyAccession Study accession
+ * @param {String} cluster Name of cluster, as defined at upload
+ * @param {String} annotation Full annotation name, e.g. "CLUSTER--group--study"
+ * @param {String} subsample Subsampling threshold, e.g. 100000
+ * @param {String} consensus Statistic to use for consensus, e.g. "mean"
+ *
+ * Example:
+ * https://localhost:3000/single_cell/api/v1/studies/SCP56/clusters/Coordinates_Major_cell_types.txt?annotation_name=CLUSTER&annotation_type=group&annotation_scope=study
+ */
+export async function fetchScatter(
```

Comment:
same as above, I would call this fetchStudyClusterData.  It's up to the UI whether it gets turned into a scatter plot or a nice spreadsheet, so let's make the API function neutral :)

Reply:
That works -- I went with a concisely mirrored name, see b24e07f.